src/main/java/io/split/android/client/cache/SplitCache.java

```diff
@@ -63,13 +63,13 @@ private String getChangeNumberFileName() {
     }
 
     @Override
-    public boolean addSplit(Split split) {
+    synchronized public boolean addSplit(Split split) {
```

Comment:
this will block the reader. Is there a way that the structures are nonblocking/atomic instead of synchronizing the method?

Reply:
discussed in person